app/mailers/consent_form_mailer.rb

```diff
@@ -19,86 +19,17 @@ def confirmation_refused(consent_form)
 
   def opts(consent_form)
     @consent_form = consent_form
+    @patient = consent_form
```

Comment:
This seems like a point of confusion to me. We have a `Patient` model so calling a `ConsentForm` "patient" doesn't feel right. I'm also not quite convinced about putting everything in `ApplicationMailer` here, tbh. I feel like it'd be fine, and easier to understand, if `personalisation` was defined in the mailer file here.

Reply:
It's because some of the fields in ApplicationMailer rely on `@patient`, but `@patient` and `@consent_form` share many fields. It's not perfect but the alternative is to potentially duplicate a lot of code

Reply:
I wonder if there's a better name than `@patient`, though. `@patient_or_consent_form` is longer but would be clearer. `@object` or `@record` is a bit abstract but would at least avoid the trap of assuming it's a `Patient` record ... ugh. Naming is hard.

But what I was saying is that I think DRYing makes it less clear. I think it would be better to build `personalisation` in this file, and DRY things up by using help methods (where needed). For example (note `short_date`, `short_patient_name` and `short_patient_name_apos`):

```
  def personalisation
    {
      full_and_preferred_patient_name: ...,
      location_name: @session.location.name,
      long_date: @session.date.to_fs(:app_short_date),
      observed_session: ...,
      parent_name: @consent_form.parent_name,
      short_date: @session.date.to_fs(:app_short_date),
      short_patient_name: short_patient_name(@consent_form),
      short_patient_name_apos: short_patient_name_apos(@consent_form),
      team_email: @session.team.email,
      team_phone: @session.team.phone,
      vaccination: ...
    }
  end
```

Reply:
I see your point, I think your suggestion would work as well. I considered something like that, but thought I'd try the Superdry (TM) version first. Let's try this version for now and if it's clunky we can roll forward to a wetter medium.